thetis/solver2d.py

```diff
@@ -834,33 +835,52 @@ def load_state(self, i_export, outputdir=None, t=None, iteration=None):
         for e in self.exporters.values():
             e.set_next_export_ix(self.i_export + offset)
 
-    def print_state(self, cputime):
+    def print_state(self, cputime, print_header=False):
         """
         Print a summary of the model state on stdout
 
-        :arg float cputime: Measured CPU time
+        :arg float cputime: Measured CPU time in seconds
+        :kwarg print_header: Whether to print column header first
         """
+        # generate simulation time string
+        if self.options.simulation_initial_date is not None:
+            now = self.options.simulation_initial_date + datetime.timedelta(seconds=self.simulation_time)
+            time_str = f'{now:%Y-%m-%dT%H:%M:%S%Z}'
+        else:
+            time_str = f'{self.simulation_time:.2f}'.rjust(22)
+
+        # list of entries to print: (header, value, format)
+        entries = [
+            ('exp', self.i_export, '5d'),
+            ('iter', self.iteration, '5d'),
+            ('time', time_str, '22s'),
+        ]
+
         if self.options.tracer_only:
             for label in self.options.tracer:
                 norm_q = norm(self.fields[label])
-
-                line = ('{iexp:5d} {i:5d} T={t:10.2f} '
-                        '{label:16s}: {q:10.4f} {cpu:5.2f}')
-
-                norm_label = label if len(label) < 3 or label[-3:] != '_2d' else label[:-3]
-                print_output(line.format(iexp=self.i_export, i=self.iteration,
-                                         t=self.simulation_time,
-                                         label=norm_label + ' norm',
-                                         q=norm_q, cpu=cputime))
+                e = (label, norm_q, '10.4f')
+                entries.append(e)
```

Comment:
I assume you realised that this isn't the same behaviour as before. I was going to protest, but then I ran the multiple tracers demo and it actually looks better than what we had before:
```
  exp  iter                   time    bell_2d    cone_2d slot_cyl_2d   Tcpu
    0     0                   0.00     1.0119     1.0289     1.0800   0.00
    1    20                   0.21     1.0119     1.0289     1.0792   0.52
    2    40                   0.42     1.0119     1.0289     1.0788   0.35
    3    60                   0.63     1.0119     1.0289     1.0785   0.36
    4    80                   0.84     1.0119     1.0289     1.0783   0.38
    5   100                   1.05     1.0119     1.0289     1.0781   0.35
```
Bunching all tracers on the same line could get messy if people are using >5 tracer fields or using long names, but I don't think anyone is using Thetis for that right now so it probably isn't a problem.

Reply:
Yeah I'd prefer to print everything on the same line. I've been trying to keep the log output within 80 chars or so, but you're right that it's difficult to ensure in all cases.